### PR TITLE
HAL_ChibiOS: Added support for mRo Pixracer Pro flight controller

### PR DESCRIPTION
**Pull request by pkocmoud**

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/mRoPixracerPro/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+SPIDEV icm20948 SPI1 DEVID1  ICM_20948_CS    MODE3  2*MHZ  4*MHZ
+SPIDEV icm20608 SPI1 DEVID2  ICM_20602_CS    MODE3  2*MHZ  8*MHZ
+SPIDEV bmi088_g SPI5 DEVID1  BMI088_GYRO_CS  MODE3 10*MHZ 10*MHZ
+SPIDEV bmi088_a SPI5 DEVID2  BMI088_ACCEL_CS MODE3 10*MHZ 10*MHZ
+
+# Now some defines for logging and terrain data files.
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+# allow to have have a dedicated safety switch pin
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/mRoPixracerPro/hwdef.dat:231]: just leave this out

File: libraries/AP_HAL_ChibiOS/hwdef/mRoPixracerPro/hwdef.dat [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+# Now setup the default battery pins driver analog pins and default
+# scaling for the power brick.
+define HAL_BATT_VOLT_PIN 2
+define HAL_BATT_CURR_PIN 3
+define HAL_BATT_VOLT_SCALE 10.1
+define HAL_BATT_CURR_SCALE 17.0
+
+# No hardware safety pin, CAN safety switch only
+define HAL_HAVE_SAFETY_SWITCH 0
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/mRoPixracerPro/hwdef.dat:245]: remove this too

**Comment by tridge** [comment]
rebased and squashed

**Comment by rmackay9** [comment]
This is included in Copter-4.0.4-rc2

**Pull request merged**